torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -2254,6 +2256,14 @@ def bwd(ctx, grad, x):
             source_target="autograd.Function",
         )
 
+        # Since we speculate the fwd graph always under no grad mode, we need to
+        # set the correct grad mode (and other attributes) for the fwd output, so
+        # we do fake tensor prop on the fwd graph again.
+        fake_mode = tx.output.fake_mode
+        with mock.patch.object(fake_mode, "allow_non_fake_inputs", True):
```

Comment:
I don't want to patch, but we have to use the fake_mode from the original tracer, since the ```example_value``` of args/kwargs were generated under that fake_mode.